config/options-promise/options-promise.js

```diff
@@ -0,0 +1,42 @@
+
+
+// showing how to use esri-loader
+import {loadModules} from 'esri-loader';
+import thumbnailUrl from './thumbnail';
+export default {
+    debug: true,
+    viewOptions: {
+        center: [-85.05019999999857, 33.12552399999943],
+        zoom: 6
+    },
+    mapOptions: {
+        basemap: 'gray'
+    },
+    widgets: [{
+        type: 'esri',
+        parent: 'view',
+        position: 'top-right',
+        path: 'esri/widgets/BasemapToggle',
+        optionsPromise: new Promise((resolve) => {
```

Comment:
Unless I'm missing something, you should be able to replace this line and the next w/ 

```js
optionsPromise: loadModules(['esri/Basemap', 'esri/layers/WebTileLayer'])`
```

Then below replace the this line `resolve({` w/ `return {`

Finally remove the now extra `})` and I _think_ it should work the same as what you have.

Reply:
Yep! You're right, it works with the return. Do you think that is a good pattern to use? `optionsPromise`.... another option I had considered was a `getOptions` function which is a function returning a Promise or Object